### PR TITLE
chore(build): Bump Typescript to 3.9.2

### PR DESCRIPTION
**Pull request by GordonSmith**
Generate type files for older typescript compilers

Fixed issues relating to:  https://github.com/microsoft/TypeScript/pull/36696

Signed-off-by: Gordon Smith <GordonJSmith@gmail.com>

### CODE REVIEW

File: packages/datastore/package.json [hunk 1]
```diff
@@ -18,21 +18,30 @@
...
-    "types/*"
+    "types/*",
+    "types-3.4/*"
   ],
   "main": "dist/index.js",
   "module": "dist/index.es6",
   "unpkg": "dist/index.min.js",
   "jsdelivr": "dist/index.min.js",
   "types": "types/index.d.ts",
+  "typesVersions": {
```
Comment (jasongrout) [inline packages/datastore/package.json:29]: Wow, I didn't know this was a thing! https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-1.html#version-selection-with-typesversions

**Comment by GordonSmith** [comment]
Looks like the documentation generation is failing (typedoc probably needs a version bump).  I will take a look in the morning.

File: packages/datastore/package.json (hunk 1 above)
Comment (GordonSmith) [reply packages/datastore/package.json:29]: Been there a long time (which is good as its the type of thing you only need way after its been released!)

**Comment by GordonSmith** [comment]
@jasongrout Travis passing now...

**Comment by blink1073** [comment]
@GordonSmith this picked up conflicts from the release I did on Saturday.

**Comment by GordonSmith** [comment]
@blink1073 - rebased...

**Comment by blink1073** [comment]
Thanks again!

**Pull request merged**